nixio/data_frame.py

```diff
@@ -101,40 +109,44 @@ def write_rows(self, rows, index=None):
             self._write_data(cr_list, sl=index)
 
     def read_rows(self, index):
+        if hasattr(index, '__iter__'):
```

Comment:
It's best to check if it's an instance of Iterable than to check if it has an attribute.

In general though, I'm not sure of the purpose here. Is it for cases where the user supplies a list or tuple of indices? Would it work in that case?